src/main/scala/com/mozilla/telemetry/streaming/EventsToAmplitude.scala

```diff
@@ -204,10 +218,11 @@ object EventsToAmplitude extends StreamingJobBase {
     implicit val sc = spark.sparkContext
 
     datesBetween(opts.from(), opts.to.get).foreach { currentDate =>
-      val dataset = com.mozilla.telemetry.heka.Dataset("telemetry")
+      val dataset = com.mozilla.telemetry.heka.Dataset(config.source)
+      val topLevelFields = TOP_LEVEL_PING_FIELDS(config.source)
 
       val pings = config.getBatchFilters.filter{
-          case(name, _) => TOP_LEVEL_PING_FIELDS.contains(name)
+          case(name, _) => topLevelFields.contains(name)
```

Comment:
Oh, this is kind of silly duplicated code -- let's fix this so topLevelFilters is correctly formatted

Reply:
I don't have the context to understand why we need to translate from `-` to `_`. Are there some contexts where we want the dashed form. Is it safe to always do the replacement?

Reply:
Oh, shoot, you're right -- on disk (e.g. reading in batch pings) the docType is normalized and `-` are replaced with `_`, but in the ping fields it's still a dash